contrib/docker/entrypoint.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+set -e
+
+# Input validation
+if [[ -z "$HOST_UID" ]]; then
+    echo "ERROR: please set HOST_UID" >&2
+    exit 1
+fi
+if [[ -z "$HOST_GID" ]]; then
+    echo "ERROR: please set HOST_GID" >&2
+    exit 1
+fi
+
+# Modify the uid and gid for builder user
+groupmod --gid "$HOST_GID" builder
+usermod --uid "$HOST_UID" builder
```

Comment:
This will fail if `HOST_GID` or `HOST_UID` happens to match another gid or uid existing on the system (that happens on my local system for my gid). My suggestion will delete the conflicting user and/or group so that the `usermod`/`groupmod` commands will succeed.
```suggestion
groupmod --gid "$HOST_GID" builder || { groupdel -f $(getent group "$HOST_GID" | cut -d: -f1) && groupmod --gid "$HOST_GID" builder }
usermod --uid "$HOST_UID" builder || { userdel $(id -nu "$HOST_UID") && usermod --uid "$HOST_UID" builder }
```